lib/reverse_sentence.rb

```diff
@@ -1,6 +1,73 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
+# Time complexity: O(n2)
+# Space complexity: O(n)
+def reverse_sentence(my_sentence) 
+  #Step_0: Screen out all sentences that doesn't need to be/can't be reversed
+  if my_sentence == nil || my_sentence == "" || my_sentence.include?("\s") == false
+    return my_sentence
+    exit
+  end
+
+  #Step_1: Reverse all characters within the string in place
+  sentence_pairs = pairs_helper(my_sentence.length)
+  i = 1
+  (sentence_pairs).times do
+    temp = my_sentence[i-1]
+    my_sentence[i-1] = my_sentence[-i]
+    my_sentence[-i] = temp
+    i += 1
```

Comment:
Notice that you're reversing a couple of times in this method.  Could you instead make a helper method.  That would dry this out.